### ISSUE TITLE
Debian repo is broken for Nginx 1.27.0

### ISSUE THREAD

**Issue by n-rodriguez** [nginx/nginx#261]
Hi there!

I think there is something wrong in Debian repos : I tried to mirror Nginx repo with https://github.com/aptly-dev/aptly and get this error :

```sh
ERROR: unable to update: download errors:
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-xslt/nginx-module-xslt_1.27.0.orig.tar.gz: size check mismatch 1244887 != 1246820
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-otel/nginx-module-otel_1.27.0+0.1.0.orig.tar.gz: size check mismatch 1244887 != 1246820
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-njs/nginx-module-njs_1.27.0+0.8.4.orig.tar.gz: size check mismatch 1244887 != 1246820
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-perl/nginx-module-perl_1.27.0.orig.tar.gz: size check mismatch 1244887 != 1246820
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-geoip/nginx-module-geoip_1.27.0.orig.tar.gz: size check mismatch 1244887 != 1246820
  http://nginx.org/packages/mainline/debian/pool/nginx/n/nginx-module-image-filter/nginx-module-image-filter_1.27.0.orig.tar.gz: size check mismatch 1244887 != 1246820
```

After some investigations I found that it comes from the `Source` file (http://nginx.org/packages/mainline/debian/dists/bullseye/nginx/source/Sources) :

```sh
Format: 3.0 (quilt)
Package: nginx-module-perl
Binary: nginx-module-perl, nginx-module-perl-dbg
Architecture: any
Version: 1.27.0-1~bullseye
Maintainer: NGINX Packaging <nginx-packaging@f5.com>
Homepage: https://nginx.org/
Standards-Version: 4.1.4
Build-Depends: debhelper (>= 11), dpkg-dev (>= 1.16.1~), quilt (>= 0.46-7~), lsb-release, libssl-dev (>= 1.0.1), libpcre2-dev, zlib1g-dev, libperl-dev
Package-List:
 nginx-module-perl deb httpd optional arch=any
 nginx-module-perl-dbg deb debug optional arch=any
Directory: pool/nginx/n/nginx-module-perl
Checksums-Sha1:
 03d63568f64163c2e88b83bfdf6b5ab68f853907 1656 nginx-module-perl_1.27.0-1~bullseye.dsc
 13f8f915966217c9de583705759f4d4f74d0bf74 1246820 nginx-module-perl_1.27.0.orig.tar.gz # HERE
 19a36642829473e58fcbe93cf2271a3a0bf543a5 9064 nginx-module-perl_1.27.0-1~bullseye.debian.tar.xz
Checksums-Sha256:
 463735e9a9430c1c13af1228ce495f3165ff3c4f2cd4e6f245a72e8550a2f367 1656 nginx-module-perl_1.27.0-1~bullseye.dsc
 47cfe9d878694e553c1beff3db439dadb5fe3b30318be6062bf71d4f2e8df788 1246820 nginx-module-perl_1.27.0.orig.tar.gz
 ba0329284444ec5ad0b371b20a89ea28ab8a45f560771468f2cc2f922dc54648 9064 nginx-module-perl_1.27.0-1~bullseye.debian.tar.xz
Files:
 f2689dbacc34c15eca0294f7e77596fb 1656 nginx-module-perl_1.27.0-1~bullseye.dsc
 a50c8dbc4a84bebe6ccb00bbc8729539 1246820 nginx-module-perl_1.27.0.orig.tar.gz
 98a61c7c8c243b97fcf1d51cb28e2c73 9064 nginx-module-perl_1.27.0-1~bullseye.debian.tar.xz
```

```sh
Format: 3.0 (quilt)
Package: nginx-module-perl
Binary: nginx-module-perl, nginx-module-perl-dbg
Architecture: any
Version: 1.27.0-2~bullseye
Maintainer: NGINX Packaging <nginx-packaging@f5.com>
Homepage: https://nginx.org/
Standards-Version: 4.1.4
Build-Depends: debhelper (>= 11), dpkg-dev (>= 1.16.1~), quilt (>= 0.46-7~), lsb-release, libssl-dev (>= 1.0.1), libpcre2-dev, zlib1g-dev, libperl-dev
Package-List:
 nginx-module-perl deb httpd optional arch=any
 nginx-module-perl-dbg deb debug optional arch=any
Directory: pool/nginx/n/nginx-module-perl
Checksums-Sha1:
 29e5bab54856a240e80de482ecd5341bd7fb64f2 1656 nginx-module-perl_1.27.0-2~bullseye.dsc
 2bd0713ba75a746a72dcfdb59c67ace746d23cf8 1244887 nginx-module-perl_1.27.0.orig.tar.gz # HERE
 e793c8398db999008f941da919312c8ff9afd34a 9080 nginx-module-perl_1.27.0-2~bullseye.debian.tar.xz
Checksums-Sha256:
 eb674fde23ef3cab1d224a847b1a8bc7c80184e858d7fd72b9e1a66cb461c70e 1656 nginx-module-perl_1.27.0-2~bullseye.dsc
 b7230e3cf87eaa2d4b0bc56aadc920a960c7873b9991a1b66ffcc08fc650129c 1244887 nginx-module-perl_1.27.0.orig.tar.gz
 1cdfbc9e098271080960b263b58a4a07268deeebf4b7da694497be63a4c5a5fd 9080 nginx-module-perl_1.27.0-2~bullseye.debian.tar.xz
Files:
 ee02ff3958d9cfa1530b801fefd1e4b2 1656 nginx-module-perl_1.27.0-2~bullseye.dsc
 a1b1cbaed987893c63cf6ab23809d85f 1244887 nginx-module-perl_1.27.0.orig.tar.gz
 8fbb9c783c5f86b1a6465439ea5a385a 9080 nginx-module-perl_1.27.0-2~bullseye.debian.tar.xz
```

The same file is defined twice with a different size :

```sh
13f8f915966217c9de583705759f4d4f74d0bf74 1246820 nginx-module-perl_1.27.0.orig.tar.gz # HERE
2bd0713ba75a746a72dcfdb59c67ace746d23cf8 1244887 nginx-module-perl_1.27.0.orig.tar.gz # HERE
```

Now when I try to rebuild the nginx package I got this error :

```sh
dpkg-source -x nginx-module-perl_1.27.0-1~bullseye.dsc
gpgv: unknown type of key resource 'trustedkeys.kbx'
gpgv: keyblock resource '/tmp/dpkg-verify-sig.uyLzOeL0/trustedkeys.kbx': General error
gpgv: Signature made Tue 28 May 2024 10:35:18 PM CEST
gpgv:                using RSA key ABF5BD827BD9BF62
gpgv: Can't check signature: No public key
dpkg-source: warning: failed to verify signature on ./nginx-module-perl_1.27.0-1~bullseye.dsc
dpkg-source: error: file ./nginx-module-perl_1.27.0.orig.tar.gz has size 1244887 instead of expected 1246820
```

Thank you!

**Comment by n-rodriguez** [comment]
From https://github.com/aptly-dev/aptly/issues/946#issuecomment-2416761224 :

> Fiiles to be mirrored should have unique names, sizes and checksums, otherwise they cannot be mirrored.

**Comment by thresheek** [comment]
Yeah we had a bit of an issue when publishing 1.27.0/1.26.1 packages - https://mailman.nginx.org/pipermail/nginx/2024-August/DITLQ7X2JECNGPD47HLECKDHNVZYBRNK.html

I'll take a look if we can fix the repo not to mention those wrong sizes.

(Also, you probably need to import the signature as per https://nginx.org/en/linux_packages.html#Debian to be able to rebuild the packages)

**Comment by thresheek** [comment]
Also, we've fixed that particular issue with 1.27.0-1 in 1.27.0-2 via https://github.com/nginx/pkg-oss/commit/9e93e52cabb60e19c8dbdbd9a5f05db29a5cb604

**Comment by thresheek** [comment]
I'm not actually sure what we can do here..  The 1.27.0-1 .dsc mention an older, wrong checksum/filesize, which are correct for 1.27.0-2 .dsc and packages.
